packages/sync/index.js

```diff
@@ -55,6 +55,24 @@ module.exports = function init (config) {
     })
   }
 
+  function test () {
+    return api('GET', undefined)
+      .then(res => {
+        if (res.status === 404) {
```

Comment:
shopify's error handling leaves something to be desired.

They return an invalid 404 response with the header `Content-Type: application/json` set but an empty body, so calling `res.json()` immediately errors.

For status 401 they return 

```
{"errors": "[API] Invalid API key or access token (unrecognized login or wrong password)"}
```

I wanted to just check `if (!res.ok)` but the 404 case was special